Fix error #103 when inserting image from web

hashhar opened the pull request:
This is a work in progress relating to #103

hashhar commented:
I am kind of stuck now. I want to show a dialog in that `catch` block but I can't roll out my own dialog box using `DisplayMessage.Show()`

pekiZG commented:
@hashhar Did you manage to get your own text to show up in the message box?

If you want to change the buttons, you have enums to do that.
https://msdn.microsoft.com/en-us/library/system.windows.forms.messageboxbuttons(v=vs.110).aspx

I will have time this evening to take a look and help (GMT+1).

NattyNarwhal commented:
You could set the default button to No, so they have to actually click the button, instead of mashing enter into an exception.

hashhar commented:
I got this to work. Slight change in program logic though.

When the dialog comes up now, it will ask you if you want to go back and correct the URL. If you say yes, you can go back. If you say no, the image insertion never occurs.

I removed the try...catch sequence too because this logic makes sure you can't get to that region of code.

I'll be pushing the commits soon. Please discuss about the program logic too because I changed it and the people in-charge should agree with it.

Giorgi commented on src/managed/OpenLiveWriter.PostEditor/PostHtmlEditing/ImageEditing/ImageEditingPropertyHandler.cs:
```diff
@@ -93,8 +95,16 @@ public static ImagePropertiesInfo GetImagePropertiesInfo(IHTMLImgElement imgElem
+                {
+                    info = new ImagePropertiesInfo(new Uri(imgSrc), new Size(width, height), remoteImageDecoratorsList);
+                }
+                catch (UriFormatException)
```
I think it's better to use Uri.TryCreate instead of using try catch to detect invalid uris.

hashhar commented:
On it @Giorgi. Also, somebody would have to localise the other strings too I guess?

hashhar commented:
@Giorgi Would the generated URI be Absolute or Relative or indeterminate?

Giorgi commented:
@hashhar new Uri() default is Absolute so I guess we can use Absolute here too.

hashhar commented:
I have tested this quite rigorously on my end. I think somebody should now go ahead and review it?

**Correct me if I'm but there are no translations available yet?**

martinwoodward commented:
Thanks for the work on this guys.  What do you think to selecting the contents of the URL box and setting the focus in that box if the user presses the Yes button?  (Basically has the bad URL pre-selected for them to just fix)

hashhar commented:
I'll get it done. Thats a great idea @martinwoodward

hashhar commented:
@martinwoodward Sorry. It looks like I'm unable to figure out how to set the focus. Maybe somebody can guide or take a look on it once they have free time.

martinwoodward commented:
How about something like

```
//Ask if they want to go back and correct the link or just abort it altogether
if (DisplayMessage.Show(MessageId.InputIsNotUrl, this, _source.SourceImageLink) == System.Windows.Forms.DialogResult.Yes)
{
    _source.TabSelected();
    return;
}
```

hashhar commented:
That did it. Thanks @martinwoodward I'll submit a PR. But should I rebase or simply commit and push?

martinwoodward commented:
Just add a new commit and push to the same branch and it'll show up here.

Many thanks!

hashhar commented:
Done. Thanks for helping me out everybody. This will be my first open-source contribution.

rebornix commented:
@hashhar it's a pretty complete PR :+1: it would be perfect if you can rebase the commits.

hashhar commented:
Yup. Rebased and pushed.

dnfclas commented:
__@hashhar__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

hashhar commented:
@ScottIsAFool Should this be merged? :shipit:

ScottIsAFool commented:
Don't see why not, looks good to me.

hashhar commented:
Thanks. This was my first significant PR. :)
